--- src/lib/server/fetches/ranks.ts
+++ src/lib/server/fetches/ranks.ts
@@ -37,6 +37,7 @@
 		type: string;
 		ladder: keyof RankInfo['ranks'] | null;
 		rank: PointInfo;
+		insideTr: boolean;
 	} = { type: '', ladder: null, rank: { rank: 0, points: 0, region: '', name: '' } };
 
 	constructor(data: RankInfo) {
@@ -49,10 +50,7 @@
 		} else if (element.tagName === 'tr') {
 			// reset rank
 			this.context.rank = { rank: 0, points: 0, region: '', name: '' };
-			element.onEndTag(() => {
-				if (this.context.rank.name && this.context.ladder)
-					this.data.ranks[this.context.ladder].push(this.context.rank);
-			});
+			this.context.insideTr = true;
 		} else if (element.tagName === 'td') {
 			// extract info
 			const className = element.getAttribute('class');
@@ -66,6 +64,10 @@
 		} else if (element.tagName === 'img') {
 			this.context.rank.region = element.getAttribute('alt') || 'UNK';
 		}
+	}
+	async elementEnd(element: HTMLRewriterTypes.Element) {
+		if (element.tagName === 'tr' && this.context.rank.name && this.context.ladder)
+			this.data.ranks[this.context.ladder].push(this.context.rank);
 	}
 	async text(text: HTMLRewriterTypes.Text) {
 		if (this.context.type === 'ladder') {
